torchaudio/csrc/sox_utils.cpp

```diff
@@ -0,0 +1,70 @@
+#include <c10/core/ScalarType.h>
+#include <sox.h>
+#include <torchaudio/csrc/sox_utils.h>
+
+namespace torchaudio {
+namespace sox_utils {
+
+caffe2::TypeMeta get_dtype(
```

Comment:
nit: I was under the impression that we should avoid caffe2. Can you explain why it is necessary here?

Reply:
Simply that's what's returned when you call `dtype` on `torch::Tensor` class.
https://pytorch.org/cppdocs/api/classat_1_1_tensor.html#_CPPv4NK2at6Tensor5dtypeEv